sdk/keyvault/azkeys/azcrypto/crypto_client.go

```diff
@@ -97,21 +97,21 @@ func NewClient(keyURL string, credential azcore.TokenCredential, options *Client
 
 	genOptions.PerRetryPolicies = append(
 		genOptions.PerRetryPolicies,
-		shared.NewKeyVaultChallengePolicy(credential, genOptions.Transport),
+		shared.NewKeyVaultChallengePolicy(credential),
 	)
 	conn := generated.NewConnection(genOptions)
 
 	vaultURL, err := parseVaultURL(keyURL)
 	if err != nil {
-		return &Client{}, err
+		return Client{}, err
```

Comment:
The guidelines insist on constructors returning clients by reference, what's the reason to return by value here? One benefit of returning a pointer is you can do this to prevent users calling methods on an uninitialized client:
```suggestion
		return nil, err
```

Reply:
Good catch, the example shows a return by pointer which is what I first glanced at. Will update

Reply:
Still returning structs here